### PR TITLE
RD2: Enable user to start and view data migration

### PR DESCRIPTION
**Pull request by RadSFDeveloper**
Start data migration batch job when user clicks "Enable Enhanced Recurring Donations" > [Run Migration] . Display data migration batch progress. Mark migration task as completed when data migration job completes successfully.

# Critical Changes

# Changes

# Issues Closed

# New Metadata
Apex Class:
- UTIL_BatchJobProgress_CTRL
- UTIL_BatchJobService

Custom Labels:
- BatchProgressCompletedDate
- BatchProgressExtendedStatus
- BatchProgressJobItemsProcessed
- BatchProgressStatus
- BatchProgressTimeElapsed
- BatchProgressTitleRD2DataMigration
- BatchProgressTotalJobItems
- RD2_EnablementMigrationCompleteMessage

Lightning Component Bundle:
- batchProgress

# Deleted Metadata

### CODE REVIEW

**Comment by RadSFDeveloper** [comment]
**lurch: attach W-036728

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001CBuSQAW" target="_blank">W-036728</a>

File: src/classes/UTIL_BatchJobService.cls [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        @TestVisible
+        private Long calcPercentComplete(Integer jobItemsProcessed, Integer totalJobItems, String status) {
+            final Integer defaultPercentComplete = 5;
+
+            Long percentComplete = defaultPercentComplete;
+
+            if (ApexJobStatus.COMPLETED.name() == status) {
+                percentComplete = 100;
+
+            } else if (ApexJobStatus.QUEUED.name() == status) {
```
Comment (force2b) [inline src/classes/UTIL_BatchJobService.cls]: Holding should be in here as well, basically the same as Queued.

Comment (RadSFDeveloper) [reply src/classes/UTIL_BatchJobService.cls]: Thank you @force2b ! Changes have been applied.

**Comment by RadSFDeveloper** [comment]
@Julian88Tex , data migration UI is ready for testing.

**Comment by Julian88Tex** [comment]
Testing has passed with a few noted concerns on test charter for WI: https://foundation.lightning.force.com/lightning/r/agf__ADM_Work__c/a2x1E000001CBuSQAW/view

**Pull request merged**